plugins/node/opentelemetry-instrumentation-graphql/src/instrumentation.ts

```diff
@@ -390,29 +389,52 @@ export class GraphQLInstrumentation extends InstrumentationBase {
     });
   }
 
+  private getOperationType(
+    operation: graphqlTypes.DefinitionNode | undefined
+  ): string | null {
+    if (operation && operation.kind == 'OperationDefinition') {
+      return operation.operation;
+    }
+
+    return null;
+  }
+
+  private getOperationName(
+    operation: graphqlTypes.DefinitionNode | undefined,
+    processedArgs: graphqlTypes.ExecutionArgs
+  ): string | null {
+    if (operation && operation.kind == 'OperationDefinition') {
+      if (operation.name) {
+        return operation.name.value;
+      }
+    }
+    if (processedArgs.operationName) {
+      return processedArgs.operationName;
+    }
+    return null;
+  }
+
   private _createExecuteSpan(
     operation: graphqlTypes.DefinitionNode | undefined,
     processedArgs: graphqlTypes.ExecutionArgs
   ): api.Span {
     const config = this._getConfig();
 
-    const span = this.tracer.startSpan(SpanNames.EXECUTE, {});
-    if (operation) {
-      const name = (operation as graphqlTypes.OperationDefinitionNode)
-        .operation;
-      if (name) {
-        span.setAttribute(AttributeNames.OPERATION, name);
-      }
-    } else {
-      let operationName = ' ';
-      if (processedArgs.operationName) {
-        operationName = ` "${processedArgs.operationName}" `;
+    const operationType = this.getOperationType(operation);
+    const operationName = this.getOperationName(operation, processedArgs);
+
+    const span = this.tracer.startSpan(createExecuteSpanName());
```

Comment:
Generally we like to avoid changing the name of a span after creation if at all possible. You can avoid updating the name if you just do this here.

```suggestion
    const span = this.tracer.startSpan(createExecuteSpanName(operationType, operationName));
```

`createExecuteSpanName` already gracefully handles missing type and name so it isn't a problem.

This would also require updating `createExecuteSpanName` or `getOperationType` to match null/undefined types but that's easy enough.

Reply:
I have used your approach, decided on returning `undefined` as I've seen it being passed around.